### PR TITLE
feat: extends tailwind config styles

### PR DESCRIPTION
**Pull request by Sboonny**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

needed to use default fontsize in #51070, but it was overwritten by the config values, this aims to extend the values in config so it doesn't conflict with default tailwind styles

### CODE REVIEW

File: tools/ui-components/src/button/button.tsx [hunk 1]
```diff
@@ -5,7 +5,7 @@ const defaultClassNames = [
   'relative',
   'cursor-pointer',
   'inline-block',
-  'border-3',
+  'border-0',
```
Comment (Sboonny) [inline tools/ui-components/src/button/button.tsx:8]:
```suggestion
  'border-3',
```

File: tools/ui-components/tailwind.config.js [hunk 2]
```diff
@@ -14,89 +14,89 @@ module.exports = {
...
-        0: 'var(--gray00)',
-        50: 'var(--gray05)',
-        100: 'var(--gray10)',
-        150: 'var(--gray15)',
-        450: 'var(--gray45)',
-        750: 'var(--gray75)',
-        800: 'var(--gray80)',
-        850: 'var(--gray85)',
-        900: 'var(--gray90)'
+    extends: {
```
Comment (Sboonny) [inline tools/ui-components/tailwind.config.js:17]:
```suggestion
    extend: {
```

**Review by naomi-lgbt — APPROVED** [review]

**Comment by ahmaxed** [comment]
Would it be possible to use one of the defined sizes or add the newly used font to the list?

**Comment by Sboonny** [comment]
it would be tedious, I have faced similar issue while adding `border` and `outline`, I think we should remove the added values and keep the colors

**Comment by ahmaxed** [comment]
text-base seem to have similar font size and line height as our text-sm.
Initially the font sizes where added because our used font sizes did not map to tailwind's.
Additionally, It is helpful to have all of the design tokens in one place for maintainability.

**Comment by Sboonny** [comment]
> Initially the font sizes where added because our used font sizes did not map to tailwind's.
> Having all of the design tokens in one place, and enforcing our a set of sizes would improve maintainability

now there are tailwind base styles that align with our usage in `/learn`, adding it will cause confusion because contributors won't understand why some base styles work and others don't, and which value is `fcc-property` and `tw-property`.

> enforcing our a set of sizes would improve maintainability

People can easily get around this enforcement, they can add css inline styles, create `.css` files or use arbitrary-values. They are getting in my way more than helping.

---

> That is why I am wondering if extending the config would be necessary?

Nope, the modal PR can be finished without it, but I would love to not deal with this inconvenience and complication again, if it breaks something we can revert it, and update the modal PR

**Review by ahmaxed — APPROVED** [review]

**Pull request merged**